### ISSUE TITLE
Log is spamming "Failed to create a session" messages

### ISSUE THREAD

**Issue by diogosantana** [psi-probe/psi-probe#648]
WARN {http-nio-8080-exec-1} [o.s.s.w.c.HttpSessionSecurityContextRepository] Failed to create a session, as response has been committed. Unable to store SecurityContext.

We need to investigate whether this is a bug or not. Is seams a bug to me, but I'm not familiar with Spring Security.

I have a tomcat 8 only with probe. Every page hit generates some log lines like this above.

**Comment by hazendaz** [comment]
I'll check this issue out.  I saw that once and mistaked it for restarting tomcat.  I didn't look at the logs afterwards when all seemed well.  I'm no expert on spring security either.  It was trial and error getting it upgraded.  Hopefully I can get that resolved today.

**Comment by hazendaz** [comment]
Issue is with turning on secure cookie.  If using https, you won't see the issue.  If using http, sessions get recreated like crazy and number of places in the application fail to work.  I tried adding a patch to prevent spring security from creating cookies but that didn't help and was before I discovered this is the issue.  I'll do some more research to see what exactly is causing the secure cookie to behave this way.  I do expect it to create a new cookie but not 36 every time we hit a page.  Currently that is what happens under http. If you go into the web.xml and remove secure cookie setting, it goes back to functioning ok under both http and https.  This could be related to spring security or webmvc.

**Comment by hazendaz** [comment]
For time being to address this allowing us to move forwards as the result is very nasty, I'm going to submit a pull request to turn off secure cookie.  I still think there may be some underlying issues with session handling though as it appears we get 5 sessions right out the gate.  There also appear to be some issues where calling getSession is left without false indicator and as such those end up creating sessions.  Bigger question is why we are losing session to start with very quickly.  However, with secure cookie turned off, the outcome is highly stable with 5 sessions created in general right at start.  I'll keep this issue open though in the meantime for further research.

**Comment by MALfunction84** [comment]
If there was a bug for secure cookie, may want to reopen that too. Otherwise, it'll show up in the changelogs.

**Comment by hazendaz** [comment]
I'm going to back off on using it for the time being.  I am waiting on a few other things to hit master first.  Technically secure-cookie should mean nothing on http.  However, tomcat actually implemented on http.  I've seen issues on other containers such as weblogic when spring mvc was in use.  Hopefully tomorrow I'll get a little more insight into the issue when I get a chance to see a few folks that experienced that specific issue.

On the changelogs, will that show up because an issue was created before it was added or will my next pull request to back off be enough?  Coding is done and tested fully just holding on a few things.

**Comment by MALfunction84** [comment]
Any closed issues with a milestone show up in the changelog for the corresponding release, with a few exceptions:
* Issues with the "FoundIn-Snapshot" label are omitted.
* Issues with the "Type-Task" label are omitted.

It's why I'm such a stickler about issue labels. :smiley:

**Comment by hazendaz** [comment]
Thanks.  I'll find that and reopen it now :)

**Comment by hazendaz** [comment]
Pull request #765 will address prevention of excess sessions.  It won't fix everything as it's no silver bullet for the cases experience here but from prior experience it's best to not have new sessions get spawned as that generally runs a security risk.

**Comment by diogosantana** [comment]
I tested master now and it still continues to show this log entry.

**Comment by hazendaz** [comment]
Master no longer shows this issue.  However, there are still a bunch of sessions opened in psi probe.  There are 31 of them.  I'm not sure why.  They are all timed to expire and do.  Then only one is left.  I believe it's spring related but not entirely sure why it happens.  I haven't tried to go full secure yet but likely will test that again soon.

**Comment by hazendaz** [comment]
closing as https only is all we allow now.
